src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserRegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    public static final int MIN_LENGTH_PASSWORD_PARAMETER = 6;
+    public static final int MIN_LENGTH_LOGIN_PARAMETER = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserExist(user);
+        checkUserLoginLength(user);
+        checkUserPassLength(user);
+        checkUserAge(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUserExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserRegisterException(
+                "User with login " + user.getLogin() + " already exists!");
+        }
+    }
+
+    private void checkUserLoginLength(User user) {
+        if (user.getLogin() != null && user.getLogin().length() < MIN_LENGTH_LOGIN_PARAMETER) {
+            throw new UserRegisterException(
+                "User login should be at least " + MIN_LENGTH_LOGIN_PARAMETER
+                    + " characters long!");
+        }
+    }
+
+    private void checkUserPassLength(User user) {
+        if (user.getPassword() != null
```

Comment:
Same as above, you should also check if the user password is not null. If the password is null, it will throw a NullPointerException when trying to call `length()` method. For example, you can throw an exception if the password is null or its length is less than the minimum required length.

Reply:
It was done!